File: golang/268_missing_number.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func missingNumber(nums []int) int {
	sum := 0
	n := len(nums)
	for i := range nums {
		sum += nums[i]
	}
	total := (n * (n + 1)) / 2
	return total - sum
}

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	line := strings.Split(scanner.Text(), ",")
	var inp []int
	for i := 0; i < len(line); i++ {
		num, err := strconv.Atoi(line[i])
		if err == nil {
			inp = append(inp, num)
		}
	}
	fmt.Println(missingNumber(inp))
}

File: python/268_missing_number.py
from typing import List

class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return 0
        ans = 0
        for i in range(0, len(nums)+1):
            ans = ans ^ i
        for i in range(0, len(nums)):
            ans = ans ^ nums[i]
        return ans

sol = Solution()
inp = [int(x) for x in input().split(",")]
print(sol.missingNumber(inp))